src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5826,8 +5781,55 @@ namespace HeatBalanceSurfaceManager {
 
     void CalcHeatBalanceInsideSurf(Optional_int_const ZoneToResimulate) // if passed in, then only calculate surfaces that have this zone
     {
+        // Pass correct list of surfaces to CalcHeatBalanceInsideSurf2
+        bool const PartialResimulate(present(ZoneToResimulate));
 
-        // SUBROUTINE INFORMATION:
+        if (!PartialResimulate) {
+            for (int zoneNum = 1; zoneNum <= NumOfZones; ++zoneNum) {
+                ZoneWinHeatGain = 0.0;
```

Comment:
Is this assigning zero to each vector item each time this loop occurs?

Reply:
Yes. Suggestions?

Reply:
Well, I mean it looks like ZoneWinHeatGain is an array allocated to the number of zones.  Just like the other arrays here.  Assigning a zero value to the ZoneWinHeatGain array here is essentially doing an internal loop over all the zones and assigning every ZoneWinHeatGain index equal to zero.  

I would either change this line to `ZoneWinHeatGain(zoneNum) = 0.0;`, like this:

```
for (int zoneNum = 1; zoneNum <= NumOfZones; ++zoneNum) {
  ZoneWinHeatGain(zoneNum) = 0.0;   // THIS IS NOW INDEXED
  ZoneWinHeatGainRep(zoneNum) = 0.0;
  ZoneWinHeatGainRepEnergy(zoneNum) = 0.0;
  ZoneWinHeatLossRep(zoneNum) = 0.0;
  ZoneWinHeatLossRepEnergy(zoneNum) = 0.0;
}
```

... or just take all these arrays out of the loop here and assign them all to zero at once:

```
ZoneWinHeatGain = 0.0;
ZoneWinHeatGainRep = 0.0;
ZoneWinHeatGainRepEnergy = 0.0;
ZoneWinHeatLossRep = 0.0;
ZoneWinHeatLossRepEnergy = 0.0;
```

In that case, the loop would occur for each, but you would rely on some possibly optimized operations inside of each array and avoid the array lookup operations.  It would be interesting to do a test to see which is faster.

OR

You tell me why I missed the point of this operation and it's actually correct as-is. 😄

Reply:
Got it. These can all be simply `ZoneWinHeatGain* = 0.0;` and drop the for loop.  And this screams for a window surface lists. Adding that as well.